libpz/util/pz_buffer.cpp

```
@@ -0,0 +1,388 @@
+}
+
+/**
+ * @brief Move assignment operator.
+ * Uses move-and-swap idiom for safe resource transfer.
+ * @param other Buffer to assign from.
+ * @return Reference to this buffer.
+ */
+PzBuffer &PzBuffer::operator=(PzBuffer &&other) noexcept {
+  if (this != &other) {
+    PzBuffer temp(std::move(other)); // Move construct temporary
+    swap(temp);                      // Swap *this with temp
+                // temp is destroyed, releasing old resources
+  }
+  return *this;
+}
+
+/**
+ * @brief Factory method to create a buffer instance.
+ * @param flag Storage type flag.
+ * @return Unique pointer to a new PzBuffer.
+ */
+std::unique_ptr<PzBuffer> PzBuffer::create(PzBufferType flag) {
+  return std::unique_ptr<PzBuffer>(new PzBuffer(flag));
+}
+
+// TODO: [Configurable Token Delimiter]
+// Currently, buffer storage assumes words are separated by spaces.
+// In future, support custom delimiters so users can define what separates words
+// (e.g. commas, semicolons, etc). Also support searching that respects the
+// chosen delimiter (including phrase searches like "word1 word2"). This will
+// require:
+//   - Adding a parameter to input methods to specify delimiter.
+//   - Storing delimiter info in the buffer.
+//   - Updating tokenization and search logic accordingly.
+// Input Methods
+/**
+ * @brief Load a single word into the buffer.
+ * @param word Input word.
+ * @return true if successful, false otherwise.
+ */
+bool PzBuffer::load_word(const std::string &word) {
+  if (word.empty()) {
+    PzError::reportError(PzErrorType::PZ_INVALID_INPUT, "Empty word input");
+    return false;
+  }
+  try {
+    words_.push_back(word);
+    total_characters_ += words_.back().size();
+    apply_storage_flag();
+    return true;
+  } catch (const std::exception &e) {
+    PzError::reportError(PzErrorType::PZ_PROCESSING_ERROR,
+                         "Failed to load word: " + std::string(e.what()));
+    return false;
+  }
```

Do we really need these try-catch blocks? What are the errors that we are expecting to catch? If its bad_alloc then is it not better to let them propagate and let app crash as there's nothing we can really do?

---

true, i just thought to add try catch blocks in these loading functions to report error to pzcore if something goes wrong taking inspiration from pz_analysis.cpp, but true if we cant do anything about some errors, then we can let it crash